src/redist/targets/GenerateLayout.targets

```diff
@@ -288,11 +266,19 @@
       </BundledInstallerComponent>
       <ComponentToDownload Include="@(BundledLayoutComponent);@(BundledInstallerComponent)">
         <ShouldDownload Condition="!Exists('%(DownloadDestination)')">true</ShouldDownload>
+
+        <!--
+          Replaces the host portion of 'BaseUrl' with 'PrivateBaseUrl'.
+        -->
+        <PrivateBaseUrl Condition="'$(InternalBuild)' == 'true'">$(MSRCBaseURL)$([System.String]::new('%(BaseUrl)').Substring($([System.String]::new('%(BaseUrl)').IndexOf('/', 8))))</PrivateBaseUrl>
       </ComponentToDownload>
     </ItemGroup>
-  
+
+    <Error Condition="$([System.String]::new('%(ComponentToDownload.BaseUrl)').IndexOf('http')) == -1" Text="The BaseURL '%(BaseUrl)' needs to be set with the 'http[s]' prefix." />
```

Comment:
In general it seems like the logic here would be easier to properly write error-handling for inside the MSBuild task.